### PR TITLE
8298943: Missing escapes for single quote marks in compiler.properties

### PR DESCRIPTION
**Pull request by archiecobbs**
The source file `compiler.properties` contains the English message bundle for compiler warnings, errors, etc.

Since the strings in this file are format strings, any single quotes must be escaped by doubling them.

An example of a properly encoded entry is:

```
compiler.err.cant.assign.val.to.this=\
    cannot assign to ''this''
```

There are three entries that appear to be not escaped when they should be.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298943](https://bugs.openjdk.org/browse/JDK-8298943): Missing escapes for single quote marks in compiler.properties

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11712/head:pull/11712` \
`$ git checkout pull/11712`

Update a local copy of the PR: \
`$ git checkout pull/11712` \
`$ git pull https://git.openjdk.org/jdk pull/11712/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11712`

View PR using the GUI difftool: \
`$ git pr show -t 11712`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11712.diff">https://git.openjdk.org/jdk/pull/11712.diff</a>

</details>

### CODE REVIEW

**Review by jonathan-gibbons — APPROVED** [review]

**Comment by jonathan-gibbons** [comment]
@archiecobbs You need to fix the PR title.

An interesting followup might be to update the `CheckResourceKeys` test to proactively check for this error condition.

**Comment by archiecobbs** [comment]
> An interesting followup might be to update the `CheckResourceKeys` test to proactively check for this error condition.

Great idea...  and no reason not to check for overall `MessageFormat` syntactical correctness since we effectively must replicate its (simplistic & quirky) parsing algorithm to do this.

Added in c01be470ba6.

File: test/langtools/tools/javac/diags/CheckResourceKeys.java [hunk 1]
```diff
@@ -312,6 +323,109 @@ void findMissingKeys(Set<String> codeStrings, Set<String> resourceKeys) {
...
+            }
+        }
+    }
+
+    /**
+     * Do some basic validation of a {@link java.text.MessageFormat} format string.
+     *
+     * <p>
+     * This checks for balanced braces and unnecessary quoting.
+     * Code cut, pasted, &amp; simplified from {@link java.text.MessageFormat#applyPattern}.
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/diags/CheckResourceKeys.java:345]: I'm guessing you copied it, not cut it .... :-)

**Review by jonathan-gibbons — APPROVED** [review]
Thanks for adding in the check.

At some point (not here and now in this review) we should maybe try and merge this with the copy-pasted version for javadoc, also called CheckResourceKeys.

**Comment by archiecobbs** [comment]
> At some point (not here and now in this review) we should maybe try and merge this with the copy-pasted version for javadoc, also called CheckResourceKeys.

Agreed... and to expand/digress even further, what would really be nice is for `MessageFormat` to make its double secret parsing algorithm (which supports arbitrarily many levels of nesting) available to the general public somehow. This would make it easier to develop tools related to format strings (see also [this class](https://github.com/archiecobbs/dellroad-stuff/blob/master/dellroad-stuff-main/src/main/java/org/dellroad/stuff/text/MessageFmt.java) which relies on introspection hacks).

Thanks for the review.

/integrate

**Comment by jonathan-gibbons** [comment]
> > At some point (not here and now in this review) we should maybe try and merge this with the copy-pasted version for javadoc, also called CheckResourceKeys.
> 

A different JDK-specific place to put the format checks might be the properties compiler used in the JDK build to convert resource files from .properties format to compiled Java classes.

**Pull request closed without merging**